export/securedrop_export/print/print_dialog.py

```diff
@@ -0,0 +1,95 @@
+        self.connect("activate", self.on_activate)
+
+    def on_activate(self, app):
+        """On GUI startup"""
+        window = Gtk.Window(application=app)
+        self.dialog = Gtk.PrintUnixDialog.new("Print Document", window)
+        self.dialog.connect("response", self.on_response)
+        self.dialog.show()
+        window.hide()
+
+    def on_response(self, parent_widget, response_id):
+        """
+        When the user clicks one of the print dialog's action buttons
+        [ CANCEL ] [ PRINT ]
+        """
+
+        if response_id == Gtk.ResponseType.OK:  # Print
+            settings = self.dialog.get_settings()
+            printer = self.dialog.get_selected_printer()
+            page_setup = self.dialog.get_page_setup()
+
+            # Fix upstream GTK page ranges:
+            #   - not properly passed to IPP printers
+            #   -
+            page_range_str = ""
+            page_ranges = settings.get_page_ranges()
+            if len(page_ranges) > 1:
+                self.show_error_disallowed_range()
+                return
+            elif len(page_ranges) == 1:
+                start = page_ranges[0].start + 1  # start at 0
+                end = page_ranges[0].end + 1  # start at 0
+                if start == end:
+                    page_range_str = str(start)
+                else:
+                    page_range_str = f"{start}-{end}"
+                settings.set("cups-page-ranges", page_range_str)
+
+            self.dialog.hide()
+            job = Gtk.PrintJob.new("print job", printer, settings, page_setup)
+            job.set_source_file(self.file_to_print)
+            job.send(self.on_job_complete, user_data=None)
+        elif response_id == Gtk.ResponseType.APPLY:  # Preview (if available)
+            pass
+        elif response_id in (
+            Gtk.ResponseType.CANCEL,  # Cancel button
+            Gtk.ResponseType.DELETE_EVENT,  # Window closed
+        ):
+            self.quit()
+
+    def on_job_complete(self, print_job, user_data, error):
+        """
+        When print dialog sends over the data to CUPS for printing. This does
+        not necessarily mean that the file has been fully printed. At this
+        point we just want the print dialog to disappear.
+        """
+        if error:
+            # Error does not need to be communicated to the user
+            #   - notifications already show current print issues
+            #   - printer icon in tray menu shows print errors
+            #   - future print dialogs display issues with a printer
+            pass
+        self.quit()  # Close print dialog and exit GTK application
+
+    def show_error_disallowed_range(self):
+        dialog = Gtk.MessageDialog(
+            modal=True,
+            message_type=Gtk.MessageType.ERROR,
+            buttons=Gtk.ButtonsType.OK,
+            text="Page Range Limitation",
+            secondary_text="Providing multiple page ranges are not "
```

Comment:
One small grammar thing:

```suggestion
            secondary_text="Providing multiple page ranges is not "
```